athenz/resource_role.go

```diff
@@ -169,25 +201,84 @@ func resourceRoleRead(ctx context.Context, d *schema.ResourceData, meta interfac
 	return nil
 }
 
+// return true iff the members is includes in the members list
+func isMemberIncludes(member *zms.RoleMember, members []*zms.RoleMember) bool {
+	for _, m := range members {
+		if string(member.MemberName) == string(m.MemberName) {
+			return true
+		}
+	}
+	return false
+}
+
+// return all role members that are present in list1 but not in list2
+func retainUniqueMembers(list1, list2 []*zms.RoleMember) []*zms.RoleMember {
+	toReturn := make([]*zms.RoleMember, 0)
+	for _, m := range list1 {
+		if !isMemberIncludes(m, list2) {
+			toReturn = append(toReturn, m)
+		}
+	}
+	return toReturn
+}
+
+/*
+since we not allow configuring both members (deprecated) and member attributes at the same time, state changes CAN'T be one of the following:
+1. remove members from both attributes member and members.
+2. add members in both attributes member and members.
+*/
```

Comment:
I am not sure it's correct.
what about this use case for example:
1. the user has old `members` block defined with `member-1,member-2`.
2. the user is going to change the TF file to `member` block with `member-1` only

Reply:
so in this case the change in the state for members attributes is to delete member 1 and member 2  and the changes In the state for member attribute is to add member 2.

Reply:
this what will happen: we will perform a delete api call for member1, and will perform add api call for member2 (will cause override)